src/RMPI/helpers.jl

```diff
@@ -63,10 +63,10 @@ The MPI barrier with optional argument. MPI syncronizing.
 mpi_barrier(comm = mpi_comm()) = MPI.Barrier(comm)
 
 """
-    targetrank(key, np, hash = hash(key))
+    targetrank(key, np)
 Compute the rank where the `key` belongs.
 """
-targetrank(key, np, hash = hash(key)) = hash%np
+targetrank(key, np) = hash(key, hash(1)) % np
```

Comment:
The hashing is indeed deterministic. This is to prevent using the same hash for the `Dict` and for `targetrank`.

I used `hash(key, hash(1))` because it changes the whole hash; using `hash(key, 1)` only changed the last couple of digits in the cases I tried.

Reply:
Cool, thanks!